File: app/backend/trade-app/src/main/java/com/trade_app/dto/request/TradeRequest.java
package com.trade_app.dto.request;

import com.trade_app.domain.enums.DeliveryType;
import com.trade_app.domain.enums.Direction;
import com.trade_app.domain.enums.Unit;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;


@Data
@AllArgsConstructor
@NoArgsConstructor
public class TradeRequest {
    private Long instrumentId;
    private Long accountId;
    private Integer quantity;
    private Float price;
    private Direction direction;
    private DeliveryType deliveryType;
    private Unit unit;
}

File: app/frontend/trade-app/src/app/shared/models/trade-request.ts
import {Direction} from '../enums/direction';
import {Unit} from '../enums/unit';
import {DeliveryType} from '../enums/delivery-type';

export interface TradeRequest {
  instrumentId: number,
  accountId: number,
  quantity: number,
  price: number,
  direction: Direction,
  deliveryType: DeliveryType,
  unit:Unit
}
